lms/static/js/student_account/components/PasswordResetConfirmation.jsx

```diff
@@ -0,0 +1,140 @@
+      password2: '',
+      passwordsMatch: true,
+      isValid: true,
+      validationMessage: '',
+    };
+    this.onBlurPassword1 = this.onBlurPassword1.bind(this);
+    this.onBlurPassword2 = this.onBlurPassword2.bind(this);
+  }
+
+  onBlurPassword1(password) {
+    this.updatePasswordState(password, this.state.password2);
+    this.validatePassword(password);
+  }
+
+  onBlurPassword2(password) {
+    this.updatePasswordState(this.state.password1, password);
+  }
+
+  updatePasswordState(password1, password2) {
+    this.setState({
+      password1,
+      password2,
+      passwordsMatch: !password1 || !password2 || (password1 === password2),
+    });
+  }
+
+  validatePassword(password) {
+    fetch('/api/user/v1/validation/registration', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        password,
+      }),
+    })
+    .then(res => res.json())
+    .then((response) => {
+      let validationMessage = '';
+      if (response.validation_decisions && response.validation_decisions.password) {
+        validationMessage = response.validation_decisions.password;
+      }
+      this.setState({
+        isValid: !validationMessage,
+        validationMessage,
+      });
+    });
+  }
+
+  render() {
+    return (
+      <section id="password-reset-confirm-anchor" className="form-type">
+        <div id="password-reset-confirm-form" className="form-wrapper" aria-live="polite">
+          <StatusAlert
+            alertType="danger"
+            dismissible={false}
+            open={!!this.props.errorMessage}
+            dialog={this.props.errorMessage}
+          />
+
+          <form id="passwordreset-form" method="post" action="">
+            <h2 className="section-title lines">
+              <span className="text">
+                {gettext('Reset Your Password')}
+              </span>
+            </h2>
+
+            <p className="action-label" id="new_password_help_text">
+              {gettext('Enter and confirm your new password.')}
+            </p>
+
+            <PasswordResetInput
+              name="new_password1"
+              describedBy="new_password_help_text"
+              label={gettext('New Password')}
+              onBlur={this.onBlurPassword1}
+              isValid={this.state.isValid}
+              validationMessage={this.state.validationMessage}
+            />
+
+            <PasswordResetInput
+              name="new_password2"
+              describedBy="new_password_help_text"
+              label={gettext('Confirm Password')}
+              onBlur={this.onBlurPassword2}
+              isValid={this.state.passwordsMatch}
+              validationMessage={gettext('Passwords do not match.')}
+            />
+
+            <input
+              type="hidden"
+              id="csrf_token"
+              name="csrfmiddlewaretoken"
+              value={this.props.csrfToken}
+            />
+
+            <Button
```

Comment:
I comment on this in the description above. Since we check this stuff onBlur, tabbing out of the password2 input means we might go past the button before re-enabling it.

(We could do simply a visual hint that it's disabled, but still leave it enabled... But eh. I figured let them press it.)